miner/worker.go

```diff
@@ -92,7 +92,7 @@ type Result struct {
 type worker struct {
 	mu sync.Mutex
 
-	agents []Agent
+	agents map[Agent]struct{}
```

Comment:
The benefit is being able to call `delete` instead of having to use a for loop and slice operations.